uni/lib/view/academic_path/widgets/course_units_card.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:uni/generated/l10n.dart';
+import 'package:uni/model/entities/course_units/course_unit.dart';
+import 'package:uni/model/entities/profile.dart';
+import 'package:uni/model/providers/startup/profile_provider.dart';
+import 'package:uni/utils/drawer_items.dart';
+import 'package:uni/view/common_widgets/generic_card.dart';
+import 'package:uni/view/course_units/widgets/course_unit_card.dart';
+import 'package:uni/view/lazy_consumer.dart';
+
+class CourseUnitsCard extends GenericCard {
+  CourseUnitsCard({super.key});
+
+  @override
+  void onRefresh(BuildContext context) {
+    Provider.of<ProfileProvider>(context, listen: false).forceRefresh(context);
+  }
+
+  @override
+  Widget buildCardContent(BuildContext context) {
+    return LazyConsumer<ProfileProvider, Profile>(
+      builder: (context, profile) {
+        final courseUnits = profile.courseUnits
+            .where(
+              (courseUnit) =>
+                  courseUnit.enrollmentIsValid() && courseUnit.grade == '',
+            )
+            .take(5)
+            .toList();
+        return _generateCourseUnitsCards(courseUnits, context);
+      },
+      hasContent: (Profile profile) => profile.courseUnits.isNotEmpty,
+      onNullContent: Center(
+        heightFactor: 10,
+        child: Text(
+          S.of(context).no_course_units,
+          style: Theme.of(context).textTheme.titleLarge,
+        ),
+      ),
+    );
+  }
+
+  Widget _generateCourseUnitsCards(
+    List<CourseUnit> courseUnits,
+    BuildContext context,
+  ) {
+    if (courseUnits.isEmpty) {
+      return Center(
+        heightFactor: 3,
+        child: Text(
+          S.of(context).no_course_units,
+          style: Theme.of(context).textTheme.titleLarge,
+        ),
+      );
+    }
+
+    return Column(
+      children: courseUnits
+          .map(
+            (courseUnit) => Column(
+              children: [
+                CourseUnitCard(courseUnit),
+                const SizedBox(height: 10),
```

Comment:
Maybe you should use the [Padding widget](https://api.flutter.dev/flutter/widgets/Padding-class.html) instead of using a `SizedBox`. Note that you can use `EdgeInsets.symmetric` to make it even more semantic.

Reply:
@vitormpp can you look at this?